app/services/ami_cache_service.rb

```diff
@@ -0,0 +1,62 @@
+class AmiCacheService
+  def cache_ami_chart_data(units)
+    return unless units
+
+    ami_types = get_ami_chart_data_from_units(units)
+    ami_types&.each do |ami_type|
+      Force::ListingService.ami({ data: ami_type, force: true })
+    end
```

Comment:
issue: This code is calling the `ami` endpoint once for every element in the array, which is not how it is done on the frontend. We need to construct a query param string from the array, and call the endpoint once with said string. See the `getAmiCharts` function for reference.

Reply:
It builds a query string in the frontend to call our ruby backend, but then deconstructs the query string in the backend to make multiple calls to salesforce: https://github.com/SFDigitalServices/sf-dahlia-web/blob/DAH-2371-cach-ami/app/controllers/api/v1/listings_controller.rb#L61

Since we are already in the backend when we are refreshing the cache, we can skip the step of building a query string

Reply:
thanks for the correction! I missed reading the backend part of the API call